### DISCUSSION TITLE
Can you help me out with cucumber parallelization and github actions?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#17010] [Questions and Help] [answered]
Hey everyone. I have a suite ready that I'm running manually, I have the dashboard service configured, so I'm recording my runs.I'm trying to find a doc or resource to learn how can I run my tests in parallel and then integrate it with Github Actions but I got a bit lost. 

My framework uses Cucumber Preprocessor + cypress, so I'm working with feature files, not specs. 
The project I'm testing (separate repo) is using github actions to build the environment so I'd like for my project to listen to that project and run after each PR is merged.

Can you point me in the right direction? Which should be my next steps? What I want to accomplish is: 
1. Configure my suite to record runs on cypress dashboard, when a PR is merged in the main project. 
2. Execute this run in parallel to reduce execution time.

**Comment by portyu9** [comment, marked as answer]
Hi @MatiDiez 

Yes. Cucumber does not prevent Cypress parallelization. Cypress Cloud parallelizes **spec files**, so your `.feature` files can be distributed across workers as long as your Cucumber preprocessor registers them as Cypress specs.

In GitHub Actions, create multiple workers and enable recording/parallelization:

```yml
strategy:
  matrix:
    container: [1, 2, 3, 4]

steps:
  - uses: actions/checkout@v4

  - uses: cypress-io/github-action@v7
    with:
      record: true
      parallel: true
```

Store your Cypress Record Key in GitHub Secrets.

For the separate application repository, trigger the Cypress repository after the application PR is merged using a GitHub Actions `repository_dispatch` event (or run the test workflow directly from the application repository).

So the flow should be:

```text
PR merged
→ trigger Cypress workflow
→ multiple GitHub workers start
→ Cypress Cloud distributes feature/spec files
→ results recorded in Cypress Cloud
```

Please mark this answer as accepted if it helped.
